Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -1230,111 +1214,105 @@ extension SourceKitServer {
         )
       }
 
-      resultHandler(locations.isEmpty ? fallbackLocation : locations, nil)
+      completion(locations.isEmpty ? fallbackLocation : locations, nil)
     }
-    let request = Request(symbolInfo, id: req.id, clientID: ObjectIdentifier(self),
-                          cancellation: req.cancellationToken, reply: callback)
+    let request = Request(symbolInfo, id: id, clientID: ObjectIdentifier(self),
+                          cancellation: cancellation, reply: callback)
     languageService.symbolInfo(request)
   }
 
-  // FIXME: a lot of duplication with definition request
-  func implementation(
-    _ req: Request<ImplementationRequest>,
+  func definition(
+    _ req: Request<DefinitionRequest>,
     workspace: Workspace,
     languageService: ToolchainLanguageServer
   ) {
-    let symbolInfo = SymbolInfoRequest(textDocument: req.params.textDocument, position: req.params.position)
-    let index = self.workspaceForDocument(uri: req.params.textDocument.uri)?.index
-    let callback = callbackOnQueue(self.queue) { (result: Result<SymbolInfoRequest.Response, ResponseError>) in
-      guard let symbols: [SymbolDetails] = result.success ?? nil, let symbol = symbols.first else {
-        if let error = result.failure {
+    findIndexedOccurrences(
+      textDocument: req.params.textDocument,
+      position: req.params.position,
+      id: req.id,
```

Comment:
Looking at the old implementation of `definition`, I think you need to pass `useLocalFallback: true` here.

Reply:
Good catch, thanks!